src/rms/queries/VisitorTracking.java

```diff
@@ -41,6 +41,9 @@ public String getVisitorUsingBadgeId(String badge_id) {
 		return jtemp.queryForObject("SELECT first_name||' '||last_name FROM visitors WHERE badge_id = ?", String.class, badge_id);
 	}
 	
+	public int checkoutUsingVisitorId(String visitor_id){
+		return jtemp.update("UPDATE visitors SET has_checked_out = 1, checked_out_time = CURRENT_TIMESTAMP WHERE visitor_id = ?",visitor_id);
+	}
```

Comment:
we need the condition AND has_checked_out = 0, this would solve the checkout button problem in first comment.